### ISSUE TITLE
Build warnings/errors with shadow-cljs 2.4.8 and CLJS 1.10.335

### ISSUE THREAD

**Issue by p-himik** [thheller/shadow-cljs#325]
A small fragment of the initial build log (there're much more warnings about undeclared vars from multiple libraries and CLJS itself):
```
------ WARNING #136 ------------------------------------------------------------
 File: shadow/animate.cljs:62:1
 Use of undeclared Var shadow.animate/__extmap
--------------------------------------------------------------------------------

------ WARNING #137 ------------------------------------------------------------
 File: shadow/animate.cljs:62:1
 Use of undeclared Var shadow.animate/__meta
--------------------------------------------------------------------------------

------ WARNING #138 ------------------------------------------------------------
 File: shadow/animate.cljs:62:1
 Use of undeclared Var shadow.animate/__hash
--------------------------------------------------------------------------------
```

Errors from the log when I tried changing some sources:
```
------ ERROR -------------------------------------------------------------------
 File: jar:file:/home/p-himik/.m2/repository/org/clojure/core.async/0.4.474/core.async-0.4.474.jar!/cljs/core/async/impl/ioc_helpers.cljs:5:1
--------------------------------------------------------------------------------
   2 |   (:require [cljs.core.async.impl.protocols :as impl])
   3 |   (:require-macros [cljs.core.async.impl.ioc-macros :as ioc]))
   4 | 
   5 | (def ^:const FN-IDX 0)
-------^------------------------------------------------------------------------
Can't redefine a constant at line 5 cljs/core/async/impl/ioc_helpers.cljs
--------------------------------------------------------------------------------
   6 | (def ^:const STATE-IDX 1)
   7 | (def ^:const VALUE-IDX 2)
   8 | (def ^:const BINDINGS-IDX 3)
   9 | (def ^:const EXCEPTION-FRAMES 4)
--------------------------------------------------------------------------------
```

**Comment by thheller** [comment]
`335` renamed a bunch of analyzer ops. I'll adjust the extra "hacks" I do for externs inference. They might not even be required anymore since some adjustments were made in core directly.

**Comment by thheller** [comment]
Fixed in `2.4.9` which also now uses `1.10.335`.
